File: app/src/main/java/closer/vlllage/com/closer/handler/group/GroupDraftHandler.java
package closer.vlllage.com.closer.handler.group;

import closer.vlllage.com.closer.pool.PoolMember;
import closer.vlllage.com.closer.store.StoreHandler;
import closer.vlllage.com.closer.store.models.Group;
import closer.vlllage.com.closer.store.models.GroupDraft;
import closer.vlllage.com.closer.store.models.GroupDraft_;

public class GroupDraftHandler extends PoolMember {
    public void saveDraft(Group group, String message) {
        $(StoreHandler.class).getStore().box(GroupDraft.class)
                .query()
                .equal(GroupDraft_.groupId, group.getId())
                .build().subscribe().single()
                .observer(groupDrafts -> {
                    GroupDraft groupDraft;

                    if (groupDrafts.isEmpty()) {
                        groupDraft = new GroupDraft().setGroupId(group.getId()).setMessage(message);
                    } else {
                        groupDraft = groupDrafts.get(0);
                    }

                    groupDraft.setMessage(message);
                    $(StoreHandler.class).getStore().box(GroupDraft.class).put(groupDraft);
                });
    }

    public String getDraft(Group group) {
        GroupDraft draft = $(StoreHandler.class).getStore().box(GroupDraft.class)
                .query()
                .equal(GroupDraft_.groupId, group.getId())
                .build()
                .findFirst();

        if (draft != null) {
            return draft.getMessage();
        }

        return "";
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/group/GroupDraftHandler.kt
package closer.vlllage.com.closer.handler.group

import com.queatz.on.On
import closer.vlllage.com.closer.store.StoreHandler
import closer.vlllage.com.closer.store.models.GroupDraft
import closer.vlllage.com.closer.store.models.GroupDraft_
import io.objectbox.query.QueryBuilder

class GroupDraftHandler constructor(private val on: On) {
    fun saveDraft(groupId: String, message: String? = null, post: String? = null) {
        on<StoreHandler>().store.box(GroupDraft::class)
                .query()
                .equal(GroupDraft_.groupId, groupId, QueryBuilder.StringOrder.CASE_SENSITIVE)
                .build().subscribe().single()
                .observer { groupDrafts ->
                    val groupDraft: GroupDraft

                    if (groupDrafts.isEmpty()) {
                        groupDraft = GroupDraft().apply { this.groupId = groupId }
                    } else {
                        groupDraft = groupDrafts[0]
                    }

                    message?.let { groupDraft.message = it }
                    post?.let { groupDraft.post = it }

                    on<StoreHandler>().store.box(GroupDraft::class).put(groupDraft)
                }
    }

    fun getDraft(groupId: String) = on<StoreHandler>().store.box(GroupDraft::class)
            .query()
            .equal(GroupDraft_.groupId, groupId, QueryBuilder.StringOrder.CASE_SENSITIVE)
            .build()
            .findFirst()
}
